vendor-extra/ViewsBundle/src/Views/SimplifiedVisitor.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ViewsBundle\Views;
+
+use FluentDOM\DOM\Element;
+use function is_string;
+use function sprintf;
+
+trait SimplifiedVisitor
+{
+    final public function visit(Element $object, View $view, array &$context = []) : View
+    {
+        if ($this->expectedTemplate() !== $view->getTemplate()) {
+            return $view;
+        }
+
+        $expectedElement = $this->expectedElement();
+        $actualElement = sprintf('{%s}%s', $object->namespaceURI, $object->localName);
+
+        if (is_string($expectedElement) && $expectedElement !== $actualElement) {
+            return $view;
+        }
+
+        foreach ($this->unexpectedArguments() as $argument) {
```

Comment:
the unexpected arguments also seem to be in `and`: as soon as one appears the Visitor does nothing. They could be in `or` as well, but we don't have an example where we have two?

Reply:
Yeah, not entirely sure about this yet.